### PR TITLE
[FB-858] Update memcached version

### PR DESCRIPTION
**Pull request by caesarprime**
Description of your patch
-------------
The patch updates the `memcached` version of a memcached instance included environment to the latest available version.

I have added `overrides.rb` to the attributes directory of each cookbook (`memcached` and `custom-memcached`) such that modifications can be made just by setting the value through it.

Recommended Release Notes
-------------
Applies `Security Updates` to Memcached. Fixes the vulnerability `CVE-2019-11596`, `CVE-2016-8704`, `CVE-2016-8705` and `CVE-2016-8706`

Estimated risk
-------------
Normal

Components involved
-------------
`memcached` cookbook, `custom-memcached` cookbook, `ey-custom` cookbook

Description of testing done
-------------

** Memcached Cookbook **

* Booted an environment with a Ruby app , Database and a redis util instance and a memcached instance
* checked whether memcached has got any new updates by connecting to the memcached instance (emerge -upDN1 world or eix net-misc/memcached)
* Imported `memcached` via `depends 'memcached'` on ey-custom/metadata.rb, and added `import_recipe 'memcached::default'` to ey-custom/recipes/after_main.rb
* Executed `/home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
* Checked `memcached` for the latest available version - `eix net-misc/memcached` and  'monit summary` to check that monit is running for `memcached`

** Redis Custom Cookbook **

* Booted an environment with a Ruby app , Database and a redis util instance and a memcached instance
* checked whether redis has got any new updates by connecting to the memcached instance (`emerge -upDN1 world` or `eix net-misc/memcached`)
* Copied the `custom-memcached` cookbook to cookbooks directory
* Imported `custom-memcached` via `depends 'custom-memcached'` on `ey-custom/metadata.rb`, and added import_recipe 'custom-memcached::default' to ey-custom/recipes/after_main.rb
* Ran `ey-core recipes upload --environment <environment-name>`
* Run apply on environment dashboard
* Checked `memcached` for the latest available version - `eix net-misc/memcached` and  'monit summary` to check that monit is running for `memcached`

QA Instructions
-------------
* Test a PHP Application
* Test with a single instance env
* Test with a multiple instance env without memcached instance
* Test with different config envs with memcached instance

### CODE REVIEW

File: custom-cookbooks/memcached/cookbooks/custom-memcached/attributes/overrides.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+node.set['memcached']['version'] = '1.4.39-r1'
```
Comment (jfuechsl) [inline custom-cookbooks/memcached/cookbooks/custom-memcached/attributes/overrides.rb]: Why did you add this file? The version is set in the default attributes file.

File: custom-cookbooks/memcached/cookbooks/custom-memcached/attributes/default.rb [hunk 2]
```diff
@@ -7,15 +7,15 @@
 
   # Default: DO NOT install memcached
   # Override this to true to install memcached
-  memcached['perform_install'] = true
+  memcached['perform_install'] = false
```
Comment (jfuechsl) [inline custom-cookbooks/memcached/cookbooks/custom-memcached/attributes/default.rb]: Why did you change this? It is unrelated to the version update.

File: cookbooks/memcached/attributes/overrides.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+node.set['memcached']['version'] = '1.4.39-r1'
```
Comment (jfuechsl) [inline cookbooks/memcached/attributes/overrides.rb]: See comment below and in https://github.com/engineyard/ey-cookbooks-stable-v5/pull/417#pullrequestreview-278305067

File: cookbooks/memcached/attributes/default.rb [hunk 4]
```diff
@@ -15,15 +15,15 @@
   memcached['install_from_source'] = false
 
   # If you're installing from the portage tree, the latest available version is 1.4.25
-  memcached['version'] = '1.4.34'
-  memcached['download_url'] = 'https://memcached.org/files/memcached-1.4.34.tar.gz'
+  memcached['version'] = '1.4.39-r1'
+  #memcached['download_url'] = 'https://memcached.org/files/memcached-1.4.39-r1.tar.gz'
```
Comment (jfuechsl) [inline cookbooks/memcached/attributes/default.rb]: Did you test this download url? I bet it returns a 404 (Memcached does not do `-r` releases, only we do in EYGL).

File: cookbooks/memcached/attributes/default.rb [hunk 5]
```diff
@@ -15,15 +15,15 @@
...
 
   # Install memcached on a utility instance named 'memcached'
-  #memcached['install_type'] = 'NAMED_UTILS'
+  memcached['install_type'] = 'NAMED_UTILS'
```
Comment (jfuechsl) [inline cookbooks/memcached/attributes/default.rb]: Why did you change the default behaviour?

File: cookbooks/memcached/attributes/default.rb [hunk 6]
```diff
@@ -15,15 +15,15 @@
...
   memcached['utility_name'] = 'memcached'
 
   # Install memcached on all app instances, or on a solo instance
-  memcached['install_type'] = 'ALL_APP_INSTANCES'
+  # memcached['install_type'] = 'ALL_APP_INSTANCES'
```
Comment (jfuechsl) [inline cookbooks/memcached/attributes/default.rb]: And here?

File: cookbooks/memcached/attributes/overrides.rb (hunk 3 above)
Comment (caesarprime) [reply cookbooks/memcached/attributes/overrides.rb]: Removed overrides.rb

File: cookbooks/memcached/attributes/default.rb (hunk 4 above)
Comment (caesarprime) [reply cookbooks/memcached/attributes/default.rb]: Yes URL had issues. Changed the URL to work with 1.4.39

File: cookbooks/memcached/attributes/default.rb (hunk 6 above)
Comment (caesarprime) [reply cookbooks/memcached/attributes/default.rb]: Reverted the changed back to original (This had been something that I've been working to make the updates work :) ) Sorry about that

**Pull request merged**